framework/db/Connection.php

```diff
@@ -607,6 +611,7 @@ protected function initConnection()
         if ($this->charset !== null && in_array($this->getDriverName(), ['pgsql', 'mysql', 'mysqli', 'cubrid'])) {
             $this->pdo->exec('SET NAMES ' . $this->pdo->quote($this->charset));
         }
+        $this->getSupportsResultSets();
```

Comment:
You know what, I think we can hardcode this crap into the PDO class of each driver...

MSSQL (supported): https://github.com/php/php-src/blob/master/ext/pdo_dblib/dblib_stmt.c#L322
Firebird (not supported): https://github.com/php/php-src/blob/master/ext/pdo_firebird/firebird_statement.c#L723
Mysql (supported): https://github.com/php/php-src/blob/master/ext/pdo_mysql/mysql_statement.c#L933
Postgres (not supported): https://github.com/php/php-src/blob/master/ext/pdo_pgsql/pgsql_statement.c#L640
etc..

Currently not all drivers have own PDO class though.